execution_workflows/DaPars2/Dapars2_snakemake/workflow/Snakefile

```diff
@@ -0,0 +1,333 @@
+"""Snakemake pipeline for Dapars2.
+For help see: https://snakemake.readthedocs.io/en/stable/index.html.
+"""
+
+import pandas as pd
+import os
+import sys
+
+configfile:"config/config.DaPars2.yaml"
+
+samples = pd.read_csv(config["samples"]).set_index("sample", drop=False)
+
+
+#-------------------------------------------------------------------------------
+localrules: finish
+
+
+rule finish:
+    """Rule that specifies the final output.
+    """
+    input:
+        final = expand(
+            os.path.join(config["out_dir"], "merge_{sample}"),
+            sample=samples.index.values)
+        # OUT1 = os.path.join(config["out_dir"],  "DaPars2_result_chr")
+        # os.path.join(config["out_dir"], "01_DaPars2.BED")
+
+# https://evodify.com/snakemake-checkpoint-tutorial/
+#-------------------------------------------------------------------------------
+# Preprocessing: obtain suitable input formats
+rule addChrToGtf:
+    """This step adds 'chr' prefix to gtf file.
+    """
+    input:
+        gtf=config['gtf']
+    output:
+        out=os.path.join(config["out_dir"], "annotation_Chr.gtf")
```

Comment:
It seems `annotation_Chr.gtf` is not used in the workflow as an input. Can the rule be removed?

Reply:
Alternatively, as rules `addChrToGtf` and `addChrToGff3` seem to to the exact same thing, could they be merged into one rule and gtf/gff expanded by wildcards? Seems a bit messy to have two rules that do the same.